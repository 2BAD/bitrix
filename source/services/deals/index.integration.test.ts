/* tslint-env jest */
// tslint:disable: no-expression-statement object-literal-sort-keys

import Bitrix from './../../bitrix'

const WEBHOOK_URL = process.env.WEBHOOK_URL

// tslint:disable-next-line: no-if-statement
if (!WEBHOOK_URL) {
  // tslint:disable-next-line: no-throw
  throw Error('Integration tests require environmental variable `WEBHOOK_URL` to be set')
}

const { deals } = Bitrix(WEBHOOK_URL)

describe('Deals', () => {
  describe('fields', () => {
    it('should get all fields', async () => {
      const { result } = await deals.fields()
      expect(result).toMatchSnapshot()
    })
  })
})
