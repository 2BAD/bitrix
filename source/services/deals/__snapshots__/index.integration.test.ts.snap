// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deals fields should get all fields 1`] = `
Object {
  "ADDITIONAL_INFO": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Дополнительная информация",
    "type": "string",
  },
  "ASSIGNED_BY_ID": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Ответственный",
    "type": "user",
  },
  "BEGINDATE": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Дата начала",
    "type": "date",
  },
  "CATEGORY_ID": Object {
    "isDynamic": false,
    "isImmutable": true,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Направление",
    "type": "crm_category",
  },
  "CLOSED": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Закрыта",
    "type": "char",
  },
  "CLOSEDATE": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Дата завершения",
    "type": "date",
  },
  "COMMENTS": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Комментарий",
    "type": "string",
  },
  "COMPANY_ID": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Компания",
    "type": "crm_company",
  },
  "CONTACT_ID": Object {
    "isDeprecated": true,
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Контакт",
    "type": "crm_contact",
  },
  "CONTACT_IDS": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": true,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Контакты",
    "type": "crm_contact",
  },
  "CREATED_BY_ID": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": true,
    "isRequired": false,
    "title": "Кем создана",
    "type": "user",
  },
  "CURRENCY_ID": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Валюта",
    "type": "crm_currency",
  },
  "DATE_CREATE": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": true,
    "isRequired": false,
    "title": "Дата создания",
    "type": "datetime",
  },
  "DATE_MODIFY": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": true,
    "isRequired": false,
    "title": "Дата изменения",
    "type": "datetime",
  },
  "ID": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": true,
    "isRequired": false,
    "title": "ID",
    "type": "integer",
  },
  "IS_MANUAL_OPPORTUNITY": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "IS_MANUAL_OPPORTUNITY",
    "type": "char",
  },
  "IS_NEW": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": true,
    "isRequired": false,
    "title": "Новая сделка",
    "type": "char",
  },
  "IS_RECURRING": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Регулярная сделка",
    "type": "char",
  },
  "IS_REPEATED_APPROACH": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Повторное обращение",
    "type": "char",
  },
  "IS_RETURN_CUSTOMER": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Повторная сделка",
    "type": "char",
  },
  "LEAD_ID": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": true,
    "isRequired": false,
    "title": "Лид",
    "type": "crm_lead",
  },
  "LOCATION_ID": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Местоположение",
    "type": "location",
  },
  "MODIFY_BY_ID": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": true,
    "isRequired": false,
    "title": "Кем изменена",
    "type": "user",
  },
  "OPENED": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Доступна для всех",
    "type": "char",
  },
  "OPPORTUNITY": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Сумма",
    "type": "double",
  },
  "ORIGINATOR_ID": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Внешний источник",
    "type": "string",
  },
  "ORIGIN_ID": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Идентификатор элемента во внешнем источнике",
    "type": "string",
  },
  "PROBABILITY": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Вероятность",
    "type": "integer",
  },
  "QUOTE_ID": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": true,
    "isRequired": false,
    "title": "Предложение",
    "type": "crm_quote",
  },
  "SOURCE_DESCRIPTION": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Дополнительно об источнике",
    "type": "string",
  },
  "SOURCE_ID": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "statusType": "SOURCE",
    "title": "Источник",
    "type": "crm_status",
  },
  "STAGE_ID": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "statusType": "DEAL_STAGE",
    "title": "Стадия сделки",
    "type": "crm_status",
  },
  "STAGE_SEMANTIC_ID": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": true,
    "isRequired": false,
    "title": "Группа стадии",
    "type": "string",
  },
  "TAX_VALUE": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Ставка налога",
    "type": "double",
  },
  "TITLE": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": true,
    "title": "Название",
    "type": "string",
  },
  "TYPE_ID": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "statusType": "DEAL_TYPE",
    "title": "Тип",
    "type": "crm_status",
  },
  "UTM_CAMPAIGN": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Обозначение рекламной кампании",
    "type": "string",
  },
  "UTM_CONTENT": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Содержание кампании",
    "type": "string",
  },
  "UTM_MEDIUM": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Тип трафика",
    "type": "string",
  },
  "UTM_SOURCE": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Рекламная система",
    "type": "string",
  },
  "UTM_TERM": Object {
    "isDynamic": false,
    "isImmutable": false,
    "isMultiple": false,
    "isReadOnly": false,
    "isRequired": false,
    "title": "Условие поиска кампании",
    "type": "string",
  },
}
`;
